product_currency/models/product_template.py

```diff
@@ -20,16 +18,8 @@ class ProductTemplate(models.Model):
         help='Use this currency instead of the product company currency'
     )
 
-    def _product_currency(self, cr, uid, ids, name, arg, context=None):
-        res = super(ProductTemplate, self)._product_currency(
-            cr, uid, ids, name, arg, context=context)
-        for rec in self.browse(cr, uid, ids, context=context):
-            if rec.force_currency_id:
-                res[rec.id] = rec.force_currency_id.id
-        return res
-
-    _columns = {
-        'currency_id': old_fields.function(
-            _product_currency, type='many2one', relation='res.currency',
-            string='Currency'),
-    }
+    @api.multi
```

Comment:
I test it functionally and it works. But could you try to add a api.depends() decorator?

Reply:
Also, I has made a poc and when we have a multi company environment and our companies has different currency, if I  change the company of the product, the currency in product does not update. This is good or bad?

I am not sure, what do you think?